tests/unit/processing/test_process_download.py

```diff
@@ -75,7 +105,87 @@ def test_download_unknown_uri(self):
+def mock_ProcessDownload():
+    return ProcessDownload(Mock(), Mock(), Mock())
+
+
+def sample_adata(n_obs: int, n_vars: int):
+    # Create a sample AnnData object for testing
+    adata = MagicMock(spec=scanpy.AnnData, n_obs=n_obs, n_vars=n_vars)
+    return adata
+
+
+@pytest.fixture
+def mock_read_h5ad():
+    with patch("scanpy.read_h5ad") as mock_read_h5ad:
+        mock_read_h5ad.return_value = sample_adata(1, 2 * GB)
+        yield mock_read_h5ad
+
+
+# Arrange
+@pytest.mark.parametrize(
+    "adata, memory_modifier, expected",
+    [
+        (sample_adata(1, 2 * GB), 1, {"Vcpus": 1, "Memory": 4000, "MaxSwap": 0}),  # minimum memory
+        (sample_adata(1, 5 * GB), 1, {"Vcpus": 2, "Memory": 5120, "MaxSwap": 0}),  # above minimum memory
+        (sample_adata(1, 5 * GB), 1.5, {"Vcpus": 2, "Memory": 7680, "MaxSwap": 0}),  # modifier adjusted
+        (sample_adata(1, 64 * GB), 1, {"Vcpus": 16, "Memory": 64000, "MaxSwap": 300000}),  # maximum memory
+    ],
+)
+def test_estimate_resource_requirements_positive(mock_ProcessDownload, adata, memory_modifier, expected):
+    # Act & Assert
+    assert expected == mock_ProcessDownload.estimate_resource_requirements(adata, memory_modifier=memory_modifier)
+
+
+@pytest.mark.parametrize(
+    "environ,expected",
+    [
+        ({"REMOTE_DEV_PREFIX": "/stack/", "DEPLOYMENT_STAGE": "test"}, "dp-test-stack-ingest-process-fake_dataset_id"),
+        ({"REMOTE_DEV_PREFIX": "stack", "DEPLOYMENT_STAGE": "test"}, "dp-test-stack-ingest-process-fake_dataset_id"),
+        ({"DEPLOYMENT_STAGE": "test"}, "dp-test-ingest-process-fake_dataset_id"),
+    ],
+)
+def test_get_job_definion_name(mock_ProcessDownload, environ, expected):
+    # Arrange
+    with patch("os.environ", environ):
+        dataset_id = "fake_dataset_id"
+
+        # Act
+        result = mock_ProcessDownload.get_job_definion_name(dataset_id)
+
+        # Assert
+        assert result == expected
+
+
+def test_remove_prefix(mock_ProcessDownload):
+    # Act & Assert
+    assert mock_ProcessDownload.remove_prefix("prefixfake", "prefix") == "fake"
+
+
+def test_create_batch_job_definition_parameters(mock_ProcessDownload, mock_read_h5ad):
+    # Arrange
+    mock_ProcessDownload.get_job_definion_name = Mock(return_value="fake_job_definition_name")
+    mock_ProcessDownload.estimate_resource_requirements = Mock(return_value={"Vcpus": 1, "Memory": 4000, "MaxSwap": 0})
+
+    # Act
+    resp = mock_ProcessDownload.create_batch_job_definition_parameters("local_file.h5ad", "fake_dataset_id")
+
+    # Assert
+    mock_ProcessDownload.estimate_resource_requirements.assert_called_once_with(mock_read_h5ad.return_value)
+    mock_ProcessDownload.get_job_definion_name.assert_called_once_with("fake_dataset_id")
+    assert resp == {
+        "JobDefinitionName": "fake_job_definition_name",
+        "Vcpus": 1,
+        "Memory": 4000,
+        "LinuxParameters": {
+            "Swappiness": 60,
+            "MaxSwap": 0,
+        },
+    }
```

Comment:
I'm not sure this is terribly useful... sort of just testing function implementation. I would just remove it

Reply:
It might seem excessive but it establish a contract with the implementation that this is the behavior we expect. If the implementation changes so should the test.